scripts/k8s/deployer.sh

```diff
@@ -0,0 +1,192 @@
+  --workers       <num>             The number of worker nodes (Default: $WORKERS).
+  --cores         <num>             How many cores to set to each io-engine (Default: $CORES).
+  --delay                           Enable developer delayed mode on the io-engine (Default: false).
+  --disk          <size>            Add disk of this size to each worker node.
+  --dry-run                         Don't do anything, just output steps.
+  --hugepages     <num>             Add <num> 2MiB hugepages (Default: $HUGE_PAGES).
+  --label                           Label worker nodes with the io-engine selector.
+
+Command:
+  start                             Start the k8s cluster.
+  stop                              Stop the k8s cluster.
+
+Examples:
+  $(basename "$0") start --workers 2 --disk 1GiB
+EOF
+}
+
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die() {
+  local _return="${2:-1}"
+  echo_stderr "$1"
+  exit "${_return}"
+}
+
+COMMAND=
+DO_ARGS=
+while [ "$#" -gt 0 ]; do
+  case $1 in
+    -h|--help)
+      help
+      exit 0
+      shift;;
+    start)
+      COMMAND="start"
+      DO_ARGS="y"
+      shift;;
+    stop)
+      COMMAND="stop"
+      DO_ARGS="y"
+      shift;;
+    *)
+      [ -z "$DO_ARGS" ] && die "Must specify command before args"
+      case $1 in
+        --workers)
+          shift
+          test $# -lt 1 && die "Missing Number of Workers"
+          WORKERS=$1
+          shift;;
+        --cores)
+          shift
+          CORES=$1
+          shift;;
+        --disk)
+          shift
+          test $# -lt 1 && die "Missing Disk Size"
+          POOL_SIZE=$1
+          shift;;
+        --delay)
+          DELAY="true"
+          shift;;
+        --label)
+          LABEL="true"
+          shift;;
+        --hugepages)
+          shift
+          test $# -lt 1 && die "Missing hugepage number"
+          HUGE_PAGES=$1
+          shift;;
+        --dry-run)
+          if [ -z "$DRY_RUN" ]; then
+            DRY_RUN="--dry-run"
+            KIND="echo $KIND"
+            FALLOCATE="echo $FALLOCATE"
+            KUBECTL="echo $KUBECTL"
+            DOCKER="echo $DOCKER"
+          fi
+          shift;;
+        *)
+          shift;;
+      esac
+  esac
+done
+
+if [ -z "$COMMAND" ]; then
+  die "No command specified!"
+fi
+
+if [ "$COMMAND" = "stop" ]; then
+  sudo nvme disconnect-all
+  $KIND delete cluster
+  exit 0
+fi
+
+"$SCRIPT_DIR"/setup-io-prereq.sh --hugepages "$HUGE_PAGES" --nvme-tcp $DRY_RUN
+
+# Create and cleanup the tmp folder
+# Note: this is static in case you want to restart the worker node
+mkdir -p "$TMP_KIND"
+sudo rm -rf "$TMP_KIND"/*
+
+# Adds the control-plane/master node
+cat <<EOF > "$TMP_KIND_CONFIG"
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+EOF
+
+start_core=1
+nodes=()
+for node_index in $(seq 1 $WORKERS); do
+  if [ "$node_index" == 1 ]; then
+    node="kind-worker"
+  else
+    node="kind-worker$node_index"
```

Comment:
nitpick - `node="kind-worker-$node_index"`

Reply:
This is just generating the same names kind uses, and sadly we have no control over that :(
https://github.com/kubernetes-sigs/kind/issues/1401